### ISSUE TITLE
[question] Hard coding the remote_state bucket name

### ISSUE THREAD

**Issue by ozbillwang** [gruntwork-io/terragrunt#509]
Seems a simple feature, but I can't set it properly. 

s3 bucket need be global unique. So when run same codes to different aws accounts, I want to define remote_state bucket `postfix` in name, for example, `terragrunt-prod-terraform-state`

Then when apply the change, I hope the tfstat file are uploaded to `<account_id>-terragrunt-prod-terraform-state`, so I needn't hard code the bucket name directly.

how can I do that?

**Comment by brikis98** [comment]
Not sure I follow. Can you give an example?

**Comment by brikis98** [comment]
Dynamic in what sense?

**Comment by ozbillwang** [comment]
The bucket name:

     bucket         = "<account_id>-terragrunt-example-prod-terraform-state"

**Comment by brikis98** [comment]
You can try https://github.com/gruntwork-io/terragrunt#get_aws_account_id.

**Comment by ozbillwang** [comment]
Perfect, that's what I look for.

**Comment by teemuniiranen** [comment]
Seems like the link is not valid anymore and I would like to use variables there. There is also hardcoded region. What if I would like to use another region for another environment, including the state bucket? I tried to use variables but I got the following error:

`ERRO[0000] /Users/...myproject/deployments/terragrunt.hcl:8,42-45: Unknown variable; There is no variable named "env".
ERRO[0000] Unable to determine underlying exit code, so Terragrunt will exit with error code 1`

Where should be these root level terragrunt.hcl variables declared?
